src/main/kotlin/cacophony/grammars/ParseTree.kt

```diff
@@ -0,0 +1,16 @@
+package cacophony.grammars
+
+import cacophony.token.Token
+import cacophony.utils.Location
+
+sealed class ParseTree<Symbol : Enum<Symbol>> {
+    class Leaf<Symbol : Enum<Symbol>>(
+        val token: Token<Symbol>,
+    ) : ParseTree<Symbol>()
+
+    class Branch<Symbol : Enum<Symbol>>(
+        val range: Pair<Location, Location>,
```

Comment:
I think it may be part of any ParseTree (both Branch and Leaf). Leaf already has this info, but it would be easier to parse subtrees, as you don't have to branch on ParseTree type.

Reply:
ok I agree, this is not large cost to duplicate this